src/components/Table/Table.tsx

```diff
@@ -1,27 +1,85 @@
-import { Table as MantineTable, TextInput } from "@mantine/core";
-import { type ChangeEvent, type ReactNode, useMemo, useState } from "react";
-import { IconSortAscending, IconSortDescending } from "@tabler/icons-react";
+import {
+  Table as MantineTable,
+  TextInput,
+  Pagination as MantinePagination,
+  createStyles,
+  Box,
+} from "@mantine/core";
+import {
+  type ChangeEvent,
+  type ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import {
+  IconArrowBarToLeft,
+  IconArrowBarToRight,
+  IconArrowLeft,
+  IconArrowRight,
+  IconGripHorizontal,
+  IconSortAscending,
+  IconSortDescending,
+} from "@tabler/icons-react";
 import { omit } from "lodash/fp";
-import { filterData, getNextSortValue, sortData } from "../../utils/utils.ts";
+import {
+  filterData,
+  getMaximumPages,
+  getNextSortValue,
+  sortData,
+} from "../../utils/utils.ts";
+import { useElementSize } from "@mantine/hooks";
 
 export interface Column<TData> {
   columnName: string;
   exactMatch: boolean;
   header: string;
   cellRenderer?: (row: TData) => ReactNode;
+  isFilterable: boolean;
 }
 
+interface Pagination {
+  pageSize: number;
+}
 interface TableProps<TData> {
   data: TData[];
   columns: Array<Column<TData>>;
+  pagination: Pagination;
 }
 
 export type Sort = Record<string, "asc" | "desc" | null>;
 export type Filter = Record<string, string>;
 
 export function Table<T extends { id: number }>(props: TableProps<T>) {
+  const { classes } = useStyles();
+
+  const { ref: rowRef, height: rowHeight } = useElementSize();
+
+  const [windowHeight, setWindowDimensions] = useState(window.innerHeight);
+  const [numberOfRows, setNumberOfRows] = useState<number>(
```

Comment:
Why do we pass pageSize from outside, if its dynamically calculated?

Reply:
Its a default value incase you don't use the resize feature. I thought it make more sense like that. And if you resize the window than you will get the results you want per resize